distribution/lib/Standard/Table/0.2.32-SNAPSHOT/src/Data/Matching.enso

```diff
@@ -0,0 +1,167 @@
+from Standard.Base import all
+import Standard.Base.Data.Locale
+import Standard.Base.Data.Text.Regex as Regex_Module
+
+from Standard.Base.Error.Problem_Behavior as Problem_Behavior_Module import all
+
+## Strategy for matching names.
+type Matching_Strategy
+   ## UNSTABLE
+      Exact name matching.
+
+      A name is matched if its exact name is provided.
+   type Exact (case_sensitivity : (True | Case_Insensitive) = True)
+
+   ## UNSTABLE
+      Regex-based name matching.
+
+      A name is matched if its name matches the provided regular expression.
+   type Regex (case_sensitivity : (True | Case_Insensitive) = True)
+
+## UNSTABLE
+   Specifies that the operation should ignore case.
+
+   TODO: Since case-sensitive can be locale dependent, in the future this may
+         be extended with a `locale` setting.
+type Case_Insensitive
+
+
+## UNSTABLE
+   An error indicating that a criterion did not match any names in the input.
+type No_Matches_Found (criterion : Text)
+
+No_Matches_Found.to_display_text : Text
+No_Matches_Found.to_display_text =
+    "The criterion ["+this.criterion+"] did not match any names in the input."
+
+
+## UNSTABLE
+   Selects objects from an input list which match any of the provided criteria.
+
+   Arguments:
+   - objects: A list of objects to be matched.
+   - criteria: A list of texts representing the matching criteria. Their meaning
+     depends on the matching strategy.
+   - reorder: Specifies whether the matched objects should be reordered according
+     to the order of the matching criteria which matched them.
+     If it is set to False, the matched objects are returned in the same order as
```

Comment:
Aren't we using the backpack notation to refer to code in docs? I mean, should it not be written this way instead?

```
If it is set to `False`, the ma[...]
```

Reply:
Have updated the function documentation.